packages/bot-templates/src/templates/test/candle.ts

```diff
@@ -3,13 +3,20 @@ import type { IExchange } from "@opentrader/exchanges";
 import type { MarketData } from "@opentrader/types";
 import type { TBotContext } from "@opentrader/bot-processor";
 import { useMarket, useCandle, useExchange } from "@opentrader/bot-processor";
+import { BarSize } from "@opentrader/types";
 import { logger } from "@opentrader/logger";
 
 export function* testCandle(ctx: TBotContext<any>) {
   const { config: bot, onStart, onStop } = ctx;
   const exchange: IExchange = yield useExchange();
 
   if (onStart) {
+    const candles = exchange.getCandlesticks({
```

Comment:
You must `yield` promises as well. Remember this is a generator function.

```suggestion
    const candles = yield exchange.getCandlesticks({
```

Additionally due TypeScript limitation, you can type the yielded result manually:

```suggestion
    const candles: ICandlestick[] = yield exchange.getCandlesticks({
```

Reply:
Updated